services/src/main/java/io/druid/cli/convert/ConvertProperties.java

```diff
@@ -125,12 +136,111 @@
+            return ImmutableMap.of(
+                "druid.metadata.storage.connector.connectURI", properties.getProperty(PROPERTY),
+                "druid.metadata.storage.type", jdbcConnectorType
+            );
+          } else {
+            return ImmutableMap.of();
+          }
+        }
+      },
+      new PropertyConverter()
+      {
+        // Add a new coordinate for the metadata storage
+        private static final String PROPERTY = "druid.extensions.coordinates";
+        private final ObjectMapper defaultObjectMapper = new DefaultObjectMapper();
+        @Override
+        public boolean canHandle(String property)
+        {
+          return PROPERTY.equals(property);
+        }
+
+        private static final String uriPropertyKey = "druid.db.connector.connectURI";
+        @Override
+        public Map<String, String> convert(Properties properties)
+        {
+          final String jdbcUrl = properties.getProperty(uriPropertyKey);
+          if(null == jdbcUrl){
+            log.warn("No entry for [%s] found in properties! cannot add ????-metadata-storage to [%s]", uriPropertyKey, PROPERTY);
+            return ImmutableMap.of();
+          }
+
+          final String value = properties.getProperty(PROPERTY);
+          final Set<String> coordinates = Sets.newHashSet();
+          final List<String> oldCoordinates;
+          try {
+            oldCoordinates = defaultObjectMapper.readValue(
+                value, new TypeReference<List<String>>(){}
+            );
+          }
+          catch (IOException e) {
+            throw com.google.api.client.repackaged.com.google.common.base.Throwables.propagate(e);
+          }
+          coordinates.addAll(oldCoordinates);
+          coordinates.add(String.format("io.druid.extensions:%s-metadata-storage", parseJdbcUrl(jdbcUrl)));
+          try {
+            return ImmutableMap.of(PROPERTY, defaultObjectMapper.writeValueAsString(ImmutableList.copyOf(coordinates)));
+          }
+          catch (JsonProcessingException e) {
+            throw Throwables.propagate(e);
+          }
+        }
+      },
+      new ValueConverter("druid.indexer.storage.type", ImmutableMap.of("db", "metadata"))
+  );
+
+  private static enum ConversionGroup
+  {
+    FIVE_TO_SIX(converters5to6), SIX_TO_SEVEN(converters6to7);
+    private final List<PropertyConverter> converterList;
+
+    private ConversionGroup(List<PropertyConverter> converterList)
+    {
+      this.converterList = converterList;
+    }
+
+  }
+
   @Option(name = "-f", title = "file", description = "The properties file to convert", required = true)
   public String filename;
 
   @Option(name = "-o", title = "outFile", description = "The file to write updated properties to.", required = true)
   public String outFilename;
 
+  @Option(name = "-c", title = "conversionGroup", description = "The conversion group to use for converting properties", required = true, allowedValues = {
+      "FIVE_TO_SIX",
+      "SIX_TO_SEVEN"
```

Comment:
I think we should default to SIX_TO_EVEN or even remove the 0.5->0.6 code altogether